src/main/java/com/karankumar/bookproject/ui/book/BookForm.java

```diff
@@ -472,12 +471,6 @@ public void setBook(Book book) {
             return;
         }
 
-        // TODO: this should be removed. A custom shelf should not be mandatory, so it should
-        // be acceptable to the custom shelf to be null
-        if (book.getCustomShelf() == null) {
-            book.setCustomShelf(new CustomShelf("ShelfName"));
-        }
-
```

Comment:
I know this is a code smell, but I'm going to bring this back (using the `CustomShelfService`) as it's a workaround that keeps the app working for now.